InteractiveHtmlBom/web/render.js

```diff
@@ -632,6 +640,43 @@ function pointWithinDistanceToSegment(x, y, x1, y1, x2, y2, d) {
   return dx * dx + dy * dy <= d * d;
 }
 
+function modulo(n, mod) {
+  return ((n % mod) + mod ) % mod;
+}
+
+function pointWithinDistanceToArc(x, y, xc, yc, radius, startangle, endangle, d) {
+  var dx = x - xc;
+  var dy = y - yc;
+  var r_sq = dx * dx + dy * dy;
+  var rmin = Math.max(0,radius-d);
+  var rmax = (radius+d);
+
+  if (r_sq < rmin * rmin || r_sq > rmax * rmax)
+    return false;
+
+  var angle1 = modulo(startangle, 360.0);
+  var x1 = xc + radius * Math.cos(angle1);
+  var y1 = yc - radius * Math.sin(angle1);
+  var dx1 = x1 - xc;
+  var dy1 = y1 - yc;
```

Comment:
You should subtract x and y not xc and yc, we are seeking distance from (x1, y1) to (x, y)

Reply:
Yep, dumb mistake on my part.